pkg/k8sutil/portforward.go

```diff
@@ -326,14 +327,24 @@ func PortForward(localPort int, remotePort int, namespace string, getPodName fun
 
 					serviceStopCh, err := ServiceForward(clientset, cfg, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort, namespace, desiredAdditionalPort.ServiceName)
 					if err != nil {
-						log.Error(errors.Wrap(err, fmt.Sprintf("failed to execute kubectl port-forward -n %s svc/%s %d:%d", namespace, desiredAdditionalPort.ServiceName, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort)))
+						prevServiceForwardErr := prevServiceForwardErrMap[desiredAdditionalPort.ServiceName]
+						if prevServiceForwardErr == nil || prevServiceForwardErr.Error() != err.Error() {
+							log.Error(errors.Wrap(err, fmt.Sprintf("failed to execute kubectl port-forward -n %s svc/%s %d:%d", namespace, desiredAdditionalPort.ServiceName, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort)))
+						}
+						prevServiceForwardErrMap[desiredAdditionalPort.ServiceName] = err
 						continue // try again
 					}
+					
+					prevServiceForwardErrMap[desiredAdditionalPort.ServiceName] = nil
 					if serviceStopCh == nil {
 						// we didn't do the port forwarding, probably because the pod isn't ready.
 						// try again next loop
 						// The API doesn't return ports that aren't ready, so this is possibly rbac?
-						log.Error(errors.Errorf("failed to forward port. check that you have permission to run kubectl port-forward -n %s svc/%s %d:%d", namespace, desiredAdditionalPort.ServiceName, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort))
+						prevServiceForwardErr := prevServiceForwardErrMap[desiredAdditionalPort.ServiceName]
+						if prevServiceForwardErr == nil || prevServiceForwardErr.Error() != err.Error() {
```

Comment:
won't `prevServiceForwardErr` always be `nil` here since we [set it to `nil`](https://github.com/replicatedhq/kots/pull/3742/files#diff-0504f350f64aa3345543945b4fd6c1ffca0a16283faf1fd21e3c4ea92590ae6fR338) before this block?